File: backend/app/router/site.py
"""站点公开配置接口

把后端 .env 里配置的对外公开地址下发给前端：
控制台-秘钥页与模型示例面板展示的中转接口地址都来自这里，
部署到公网后只需改 backend/.env 的 PUBLIC_BASE_URL，前端自动跟随。
"""

from fastapi import APIRouter
from app.core.config import server_setting
from app.utils.response import success_response

router = APIRouter(prefix="/site", tags=["Site"])


@router.get("/info")
async def get_site_info():
    """获取站点公开配置（无需登录）：对外中转接口的基地址等"""
    return success_response(
        message="获取站点配置成功",
        data={
            "publicBaseUrl": server_setting.PUBLIC_BASE_URL.rstrip('/'),
            "relayBaseUrl": server_setting.RELAY_BASE_URL,
        },
    )

File: frontend/src/api/site.ts
// 站点公开配置接口请求（无需登录）
import request from './request'

/** 站点公开配置 */
export interface siteInfo {
    /** 对外公开的后端地址，如 http://localhost:2027 */
    publicBaseUrl: string
    /** 对外中转接口基地址（OpenAI 兼容），如 http://localhost:2027/v1 */
    relayBaseUrl: string
}

/** 获取站点公开配置（对外中转接口地址等，来源为后端 .env 的 PUBLIC_BASE_URL） */
export function getSiteInfo() {
    return request.get<siteInfo>('/site/info')
}
